File: January 17, 2026.js
function knightMoves(position) {

  const letters = ["A", "B", "C", "D", "E", "F", "G", "H"];
  const letter = position[0];
  let number = Number(position.slice(1));
  let possiblesPositions = 0;
  
  if ((number + 1 < 9) && letters.includes(String.fromCharCode(letter.charCodeAt(0) - 2))) {
    possiblesPositions += 1;
  }
  if ((number - 1 > 0) && letters.includes(String.fromCharCode(letter.charCodeAt(0) - 2))) {
    possiblesPositions += 1;
  }
  if ((number + 2 < 9) && letters.includes(String.fromCharCode(letter.charCodeAt(0) - 1))) {
    possiblesPositions += 1;
  }
  if ((number + 2 < 9) && letters.includes(String.fromCharCode(letter.charCodeAt(0) + 1))) {
    possiblesPositions += 1;
  }
  if ((number + 1 < 9) && letters.includes(String.fromCharCode(letter.charCodeAt(0) + 2))) {
    possiblesPositions += 1;
  }
  if ((number - 1 > 0) && letters.includes(String.fromCharCode(letter.charCodeAt(0) + 2))) {
    possiblesPositions += 1;
  }
  if ((number - 2 > 0) && letters.includes(String.fromCharCode(letter.charCodeAt(0) - 1))) {
    possiblesPositions += 1;
  }
  if ((number - 2 > 0) && letters.includes(String.fromCharCode(letter.charCodeAt(0) + 1))) {
    possiblesPositions += 1;
  }


  return possiblesPositions;
}

File: January 17, 2026.py
def knight_moves(position):

    possibles_positions = 0
    letters = ["A", "B", "C", "D", "E", "F", "G", "H"]
    letter = position[0] 
    number = int(position[1:])

    if number + 1 < 9 and chr(ord(letter) - 2) in letters:
        possibles_positions += 1
    if number - 1 > 0 and chr(ord(letter) - 2) in letters:
        possibles_positions += 1
    if number + 2 < 9 and chr(ord(letter) - 1) in letters:
        possibles_positions += 1
    if number + 2 < 9 and chr(ord(letter) + 1) in letters:
        possibles_positions += 1
    if number + 1 < 9 and chr(ord(letter) + 2) in letters:
        possibles_positions += 1
    if number - 1 > 0 and chr(ord(letter) + 2) in letters:
        possibles_positions += 1
    if number - 2 > 0 and chr(ord(letter) - 1) in letters:
        possibles_positions += 1
    if number - 2 > 0 and chr(ord(letter) + 1) in letters:
        possibles_positions += 1

    return possibles_positions